test/built-ins/Temporal/ZonedDateTime/prototype/with/month-and-month-code-must-agree.js

```diff
@@ -0,0 +1,16 @@
+// Copyright (C) 2021 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.zoneddatetime.prototype.with
+description: month and monthCode must agree.
+features: [Temporal]
+---*/
+
+const zdt = new Temporal.PlainDateTime(1976, 11, 18, 15, 23, 30, 123, 456, 789).toZonedDateTime("UTC");
```

Comment:
Can be any arbitrary ZonedDateTime, to avoid the test depending on the `PlainDateTime.prototype.toZonedDateTime` already being implemented:
```suggestion
const zdt = new Temporal.ZonedDateTime(0n, "UTC");
```